File: Frontend/dgu_laf/lib/model/item_comment.dart
class ItemComment {
  final int commentId;
  final int userId;
  final int itemId;
  final bool isLost;
  final String content;
  final String createdAt;

  ItemComment({
    required this.commentId,
    required this.userId,
    required this.itemId,
    required this.isLost,
    required this.content,
    required this.createdAt,
  });

  factory ItemComment.fromJson(Map<String, dynamic> json) {
    return ItemComment(
      commentId: json['comment_id'],
      userId: json['user_id'],
      itemId: json['item_id'],
      isLost: json['is_lost'],
      content: json['content'],
      createdAt: json['created_at'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'comment_id': commentId,
      'user_id': userId,
      'item_id': itemId,
      'is_lost': isLost,
      'content': content,
      'created_at': createdAt,
    };
  }
}

File: Backend/dgulaf/src/main/java/com/dgulaf/dgulaf/model/ItemComment.java
package com.dgulaf.dgulaf.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "itemcomments")
public class ItemComment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer commentId;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private Integer itemId;

    @Column(nullable = false)
    private Boolean isLost;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(nullable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime createdAt;
}
